homeassistant/components/enocean/binary_sensor.py

```diff
@@ -58,53 +63,75 @@ def device_class(self):
+    def device_type(self):
+        """Return the type of this sensor."""
+        return self._device_type
+
     def value_changed(self, packet):
         """Fire an event with the data that have changed.
 
         This method is called when there is an incoming packet associated
         with this platform.
-
         Example packet data:
         - 2nd button pressed
             ['0xf6', '0x10', '0x00', '0x2d', '0xcf', '0x45', '0x30']
         - button released
             ['0xf6', '0x00', '0x00', '0x2d', '0xcf', '0x45', '0x20']
         """
-        # Energy Bow
-        pushed = None
-
-        if packet.data[6] == 0x30:
-            pushed = 1
-        elif packet.data[6] == 0x20:
-            pushed = 0
-
-        self.schedule_update_ha_state()
-
-        action = packet.data[1]
-        if action == 0x70:
-            self.which = 0
-            self.onoff = 0
-        elif action == 0x50:
-            self.which = 0
-            self.onoff = 1
-        elif action == 0x30:
-            self.which = 1
-            self.onoff = 0
-        elif action == 0x10:
-            self.which = 1
-            self.onoff = 1
-        elif action == 0x37:
-            self.which = 10
-            self.onoff = 0
-        elif action == 0x15:
-            self.which = 10
-            self.onoff = 1
-        self.hass.bus.fire(
-            EVENT_BUTTON_PRESSED,
-            {
-                "id": self.dev_id,
-                "pushed": pushed,
-                "which": self.which,
-                "onoff": self.onoff,
-            },
-        )
+
+        if packet.data[0] == 0xF6:
+            # Energy Bow
+            pushed = None
+            pressed = None
+            first_action = None
+            second_action = None
+            second_action_valid = None
+
+            if self._device_type == 2:
+                packet.parse_eep(0x02, 0x02)
+                pressed = packet.parsed["EB"]["raw_value"]
+                first_action = packet.parsed["R1"]["raw_value"]
+                second_action = packet.parsed["R2"]["raw_value"]
+                second_action_valid = packet.parsed["SA"]["raw_value"]
+
+            if packet.data[6] == 0x30:
+                pushed = 1
+            elif packet.data[6] == 0x20:
+                pushed = 0
+
+            self.schedule_update_ha_state()
+
+            action = packet.data[1]
+            if action == 0x70:
+                self.which = 0
+                self.onoff = 0
+            elif action == 0x50:
+                self.which = 0
+                self.onoff = 1
+            elif action == 0x30:
+                self.which = 1
+                self.onoff = 0
+            elif action == 0x10:
+                self.which = 1
+                self.onoff = 1
+            elif action == 0x37:
+                self.which = 10
+                self.onoff = 0
+            elif action == 0x15:
+                self.which = 10
+                self.onoff = 1
+
+            self.hass.bus.fire(
```

Comment:
We still don't allow binary sensor entities to fire events.

If the device is stateless it should just be represented as custom events in home assistant and those should not be fired from an entity platform.

Any changes to this part won't be accepted until this is removed and reworked.

Reply:
Weel this is not a part that has been modified in the PR... It was already working like this and has not been touched.

Reply:
We're extending a pattern that we don't want. We won't accept that.

I suggest waiting with adding support for these devices until the integration is refactored.

Reply:
If I remove all changes made to binary_sensor.py, can the rest proceed ?